commit b6e13cd35e3f1925e7d505b2088822a1416081d7 (refs/changes/01/2301/2)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2017-07-31T18:25:41+00:00 (2 years, 2 months ago)
    
    OS-6256 g11n doesn't always properly clean up when running gmake clean

diff --git a/g11n/Makefile b/g11n/Makefile
index 127621c..e08db9b 100644
--- a/g11n/Makefile
+++ b/g11n/Makefile
@@ -1,24 +1,16 @@
 #
-# CDDL HEADER START
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
 #
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License (the "License").  
-# You may not use this file except in compliance with the License.
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
 #
-# You can obtain a copy of the license at src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
+
 #
-# Copyright 2013, Joyent, Inc.  All rights reserved.
+# Copyright (c) 2017, Joyent, Inc. 
 #
 
 include ../Makefile.defs
@@ -47,5 +39,12 @@ install: all
 	cp proto/i386/fileroot/usr/lib/iconv/amd64/UCS-2%UTF-8.so $(LIB64DIR)
 	cp proto/i386/fileroot/usr/lib/iconv/amd64/UCS-2LE%UTF-8.so $(LIB64DIR)
 
+#
+# We explicitly use the dmake that we install as part of the build
+# bootstrap here to invoke the clean target, as the make that we build
+# may have already been removed.
+#
 clean:
-	@(cd src; pwd; unset MAKE; SRC=$(SRC) $(SUN_MAKE) clean)
+	test -x /opt/local/bin/dmake
+	@(cd src; pwd; unset MAKE; SRC=$(SRC) /opt/local/bin/dmake clean)
+	rm -rf proto
